### PR TITLE
HD-18740 provide recovered image IDs

### PR DESCRIPTION
**Pull request by atymchuk**

### CODE REVIEW

File: vars/pmmVersion.groovy [hunk 1]
```diff
@@ -1,32 +1,51 @@
 def call(String type='latest') {
-  List<String> oldVersions = ['2.9.1', '2.10.0', '2.10.1', '2.11.0', '2.11.1', '2.12.0', '2.13.0', '2.14.0']
+  // List<String> oldVersions = ['2.9.1', '2.10.0', '2.10.1', '2.11.0', '2.11.1', '2.12.0', '2.13.0', '2.14.0']
+  List<String> oldVersions = ['2.12.0', '2.13.0']
+  List<String> oldVersions = []
```
Comment (talhabinrizwan) [inline vars/pmmVersion.groovy:4]: I don't get it. First you are setting oldVersion to '2.12.0' and '2.13.0' and then setting to an empty array?

File: vars/pmmVersion.groovy [hunk 2]
```diff
@@ -1,32 +1,51 @@
...
+    '2.12.0': 'ami-0b5539a85b450ad6a',
+    '2.13.0': 'ami-082bcd79afc35d576',
+    '2.22.0': 'ami-0fd1d4c619b968b18',
+    '2.23.0': 'ami-0a98f83f5406c379e',
+    '2.24.0': 'ami-07008195e6afe0694',
+    '2.25.0': 'ami-0b5a9dff554ea6d12',
+    '2.26.0': 'ami-09914490ee52c24c6',
+    '2.27.0': 'ami-0e1dd6bf6627aed8e',
+    '2.28.0': 'ami-0b6ea9aa3292b41dd',
+    '2.29.1': 'ami-07f8ad99be4ed007b',
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply vars/pmmVersion.groovy:37]: Do we need 2.29.0 if we have a patch version of 2.29.1?

File: vars/pmmVersion.groovy [hunk 3]
```diff
@@ -1,32 +1,51 @@
 def call(String type='latest') {
-  List<String> oldVersions = ['2.9.1', '2.10.0', '2.10.1', '2.11.0', '2.11.1', '2.12.0', '2.13.0', '2.14.0']
+  // List<String> oldVersions = ['2.9.1', '2.10.0', '2.10.1', '2.11.0', '2.11.1', '2.12.0', '2.13.0', '2.14.0']
+  List<String> oldVersions = ['2.12.0', '2.13.0']
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply vars/pmmVersion.groovy:3]: Well, we didn't recover that one :( 
Ok, will do...

[the remark this answers is not in the document]
Comment (atymchuk) [reply vars/pmmVersion.groovy:3]: oops... it turns out 2.14.0 through 2.15.1 were not published, so we can't recover those  :(

File: vars/pmmVersion.groovy (hunk 1 above)
Comment (atymchuk) [reply vars/pmmVersion.groovy]: it was a typo - fixed now.

**Review by talhabinrizwan — APPROVED** [review]

File: vars/pmmVersion.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (puneet0191) [reply vars/pmmVersion.groovy:37]: there is a possiblity someone never really upgraded to patch 2.29.1, and now they only see upgrade from 2.29.0 -> 2.34.0 or latest, hence that path needs to be validated

[the remark this answers is not in the document]
Comment (atymchuk) [reply vars/pmmVersion.groovy:41]: Done!

**Pull request merged**